lib/plugins/aws/deploy-function.js

```diff
@@ -267,6 +268,29 @@ class AwsDeployFunction {
     if (functionObj.layers && !functionObj.layers.some(_.isObject)) {
       params.Layers = functionObj.layers;
     }
+
+    let hasServerlessConsoleSDK = false;
+    let hasServerlessConsoleExternal = false;
+    const sdkRegex = /(177335420605|321667558080):layer:sls-sdk/;
+    const externalRegex = /(177335420605|321667558080):layer:sls-external-extension-/;
+    if (
+      params.Layers &&
+      remoteFunctionConfiguration.Layers &&
+      remoteFunctionConfiguration.Layers.some(
+        ({ Arn }) => sdkRegex.test(Arn) || externalRegex.test(Arn)
+      )
+    ) {
+      const serverlessConsoleLayerArns = remoteFunctionConfiguration.Layers.filter(
+        ({ Arn }) => sdkRegex.test(Arn) || externalRegex.test(Arn)
+      ).map(({ Arn }) => Arn);
+      hasServerlessConsoleSDK = serverlessConsoleLayerArns.some((layerArn) =>
+        layerArn.includes('sls-sdk')
+      );
+      hasServerlessConsoleExternal = serverlessConsoleLayerArns.some((layerArn) =>
+        layerArn.includes('sls-external-extension')
+      );
+      params.Layers.push(...serverlessConsoleLayerArns);
+    }
```

Comment:
If I read correctly this logic will add console layers, even if they're already in `params.Layers` (I know it's very unlikely, but can happen when .e.g we test SDK for new layer runtime, and add it explicitly)

Also I think we can simplify this to something as:

```javascript
const isConsoleSdkLayerArn = RegExp.prototype.test.bind(/(?:177335420605|321667558080):layer:sls-/u);

const attachedConsoleSdkLayers = (remoteFunctionConfiguration.Layers ||[]).filter(({ Arn: arn }) => isConsoleSdkLayerArn(arn));

if (attachedConsoleSdkLayers.length) {
  if (!params.Layers) params.Layers = [];
  for (const attachedConsoleSdkLayer of attachedConsoleSdkLayers) {
    if (!params.Layers.includes(attachedConsoleSdkLayer)) params.Layers.push()attachedConsoleSdkLayer
  }
}
```

Reply:
This looks good 👍 I was testing this a bit more and I think actually we need to initialize the `params.Layers` array to an empty array if it doesn't exist.

I am noticing that if you try and remove all layers from your function it will say that your configuration has not been updated because we are skipping doing anything to `params.Layers` when it doesn't exist. This appears to be an issue with or without console.

I will update this section to be much more simple as you suggested in just a minute. Thanks for the feedback as usual 😎 🙏